Framework/API/test/SampleTest.h

```diff
@@ -62,15 +62,15 @@ class SampleTest : public CxxTest::TestSuite {
   test_That_An_Environment_Can_Be_Set_And_The_Same_Environment_Is_Returned() {
     Sample sample;
     const std::string envName("TestKit");
-    SampleEnvironment *kit =
-        new SampleEnvironment(envName, boost::make_shared<const Container>(""));
+    auto kit = boost::make_shared<SampleEnvironment>(
+        envName, boost::make_shared<const Container>(""));
     kit->add(boost::make_shared<const CSGObject>());
 
     TS_ASSERT_THROWS_NOTHING(sample.setEnvironment(kit));
 
     const SampleEnvironment &sampleKit = sample.getEnvironment();
     // Test that this references the correct object
-    TS_ASSERT_EQUALS(&sampleKit, kit);
+    TS_ASSERT_EQUALS(&sampleKit, kit.get());
     TS_ASSERT_EQUALS(sampleKit.name(), envName);
     TS_ASSERT_EQUALS(sampleKit.nelements(), 2);
   }
```

Comment:
As you are already checking addresses - You should probably also check that when you create a copy of the sample the environment is the same, but after setting the sample environment on the copy, the original is not affected.